src/java/org/apache/cassandra/service/AutoRepairService.java

```diff
@@ -72,47 +80,71 @@ public void checkCanRun(RepairType repairType)
             throw new ConfigurationException("Cannot run incremental repair while CDC replay is enabled. Set cdc_on_repair_enabled to false.");
     }
 
-    @Override
     public AutoRepairConfig getAutoRepairConfig()
     {
         return config;
     }
 
     @Override
-    public void setAutoRepairEnabled(RepairType repairType, boolean enabled)
+    public boolean isAutoRepairDisabled()
     {
-        checkCanRun(repairType);
-        config.setAutoRepairEnabled(repairType, enabled);
+        return config == null || !config.isAutoRepairSchedulingEnabled();
+    }
+
+    @Override
+    public String autoRepairConfiguration()
```

Comment:
Not sure if we are supposed to do string formatting early (here) or only at the edge, i.e., inside GetAutoRepairConfig.java implementation. If the latter, then we would have to expose each attribute in the NodeProbe.java

Reply:
Yeah, discussed to Andy prior to moving this here. I think both options are viable, but we'll need to decide one way or another. The drawback of the other approach is that if you add additional properties, they will need to also be exposed through node probe

Reply:
I am perfectly fine with the current approach; I want to make sure that we are all on the same page with it, which we are.

Reply:
Yeah, this seems reasonable as is.  I think a possible argument for not doing this would be if we wanted to omit an object that NodeProbe can expose in different formats (e.g. human readable, json, etc.), but I don't think that is necessary here as I don't expect folks will be programatically parsing this anyways.  If there is a need so we can consider exposing it as a Map<String, ...> or something.

Do agree that it is nice to add here as it is less likely that someone will miss updating this when adding/changing settings.

Reply:
I think `Map<String, ...>` is a viable alternative. Maybe we can consider that as a potential follow up improvement if needed?

Reply:
Maybe we can file a P1 ticket and do it whenever time permits